docs/reflections/mueller/reflexion-sprint2.md

```diff
@@ -0,0 +1,76 @@
+
+## Sind diese Themen zum Zeitpunkt im Projekt gut gewählt, wenn ja, wieso?
+
+Grundsätzlich würde ich sagen, dass die Themen gut zu DevOps passen und definitiv relevant sind. Allerdings denke ich, dass es sinnvoll wäre, den Teil über automatisiertes Testing etwas kürzer zu halten, da wir dieses Thema bereits in einem anderen Modul (welches sich ausschliesslich mit Applikationstesting befasst) ausführlich behandelt haben.
+
+Dafür habe ich jedoch einen Themenblock zur Deployment-Infrastruktur vermisst (nicht nur mittels Docker sondern vielleicht auch noch eine weitere Technologie wie z.B. Terraform). Meiner Meinung nach ist das Deployen von Software ein zentraler Bestandteil von DevOps und extrem wichtig für den Entwicklungsprozess. Das lokale Entwickeln von Software ist meist einfach, aber eine der grössten Herausforderungen liegt darin, die Software so zu gestalten, dass sie ideal auf einem Server läuft.
+Ein zusätzlicher Block über Infrastruktur könnte hier meiner Meinung nach gut ergänzt werden. Abgesehen davon fand ich die Themenauswahl jedoch sehr gelungen.
+
+
+## Dieses Thema vom Sprint ist besonders relevant? Begründen Sie.
+Linting!. Dieses Thema wurde bisher, soweit ich weiss, in keinem anderen Schulmodul behandelt. Ich finde jedoch, dass es ein extrem wichtiges Thema ist. Anfangs dachte ich, dass es beim Linting lediglich um Syntax und das Einhalten von Formatierungen geht. Doch die praktische Anwendung hat mir gezeigt, dass Linting auch hervorragend dazu geeignet ist, Probleme zu erkennen.
+
+Viele meiner Projekte sind in Python. Da Python nicht kompiliert ist und keine Typen erzwingt, ist das Risiko von Laufzeitfehlern hoch. Automatisierte Tests können zwar einige dieser Bugs aufdecken, aber bei weitem nicht alle. Ein einfacher Tippfehler kann bereits dazu führen, dass ein schwer auffindbarer Bug entsteht. Ein typisches Beispiel: Man hat eine Liste namens `items` und iteriert über die Liste mit der Variablen `item`. Innerhalb der Schleife möchte man eine bestimmte Property auslesen, greift aber versehentlich auf `items` statt auf `item` zu.
+
+Linting kann die meisten solcher Bugs finden und so potenziell problematische Fehler frühzeitig erkennen. Daher halte ich Linting für ein extrem wichtiges Thema, das von Entwicklern oft vernachlässigt wird. Und das, obwohl es sehr schnell integriert und konfiguriert werden kann.
+
+## Was kann ich nun mehr als vorher?
+Ich kannte `DerLev/eslint-annotations@v2` vorher nicht. Zwar war mir bewusst, dass so etwas technisch möglich ist, aber ich habe nie darüber nachgedacht, den Linter direkt in eine MR zu integrieren. Bisher kannte ich es eher so, dass der Linter die Pipeline zum Absturz bringt und man anschliessend die Logs der Pipeline auslesen musste, um die Ursache des Problems zu finden.
```

Comment:
> Ich kannte `DerLev/eslint-annotations@v2` vorher nicht.

Ich auch nicht 😅 Und ich muss sagen, ich würde auch eher alle Developer dazu bringen anständige Plugins in der IDE zu haben. Nichts desto trotz können solche Tools sehr sinnvoll sein, vor allem in Open-Source Projekte auf die beliebige Leute einen PR stellen können.